### PR TITLE
Add aggregation option to reliability calibration table construction

### PR DESCRIPTION
**Pull request by tjtg**
Add an option to aggregate over coordinates as part of reliability calibration table construction.

Aggregation during construction results in a significant memory saving. Un-aggregated reliability calibration tables are quite large due to the additional table_row_index and probability_bin coordinates, in addition to existing threshold and spatial coordinates.
For example, typical values of table_row_index (3) and probability_bin (7), result in 21x the original input size. The un-aggregated reliability tables compress very well due to containing many zeroes and the resulting on-disk file is small, but it does consume a large amount of memory during processing. We have encountered approx. 60 GB memory usage on a 2.4 km resolution Australian domain.

The option added in this PR carries out aggregation as part of each threshold being processed, which significantly reduces memory usage in the common case of aggregating over spatial coordinates (eg. latitude/longitude or projection_x_coordinate/projection_y_coordinate). In the above example, memory usage reduces from 60 GB to 3.8 GB. Wall time performance is also better due to less data for the NetCDF library to pass through zlib compression (and no need to run the aggregate step separately).

Testing:
 - [x] Ran tests and they passed OK
 - [x] Added new tests for the new feature(s)

### CODE REVIEW

**Review by anja-bom — DISMISSED** [review]
Hi Tom, really nice solution to our memory problem. I made a suggestion that you can take or leave as you see fit about adding in example usage to make the CLI usage crystal clear

File: improver/cli/construct_reliability_tables.py [hunk 1]
```diff
@@ -71,6 +72,11 @@ def process(
         single_value_upper_limit (bool):
             Mandates that the highest bin should be single valued, with a small
             precision tolerance, defined as 1.0E-6. The bin is thus (1 - 1.0E-6) to 1.
+        aggregate_coordinates (List[str]):
+            An optional list of coordinates over which to aggregate the reliability
+            calibration table using summation. This is equivalent to constructing
+            then using aggregate-reliability-tables but with reduced memory
+            usage due to avoiding large intermediate data.
```
[the remark this answers is not in the document]
Comment (tjtg) [reply improver/cli/construct_reliability_tables.py:79]: The acceptance tests usually provide good examples to demonstrate usage of CLIs. I've been a bit skimpy in this PR and hadn't added a new acceptance test.
I've done that in bf44263.

**Review by benowen-bom — APPROVED** [review]
This PR presents a neat and simple solution to overcome the memory issues described. With the inclusion of acceptance test data, I think this is now complete and all looks good to me.

**Comment by MoseleyS** [comment]
Acceptance test data merged

**Comment by benowen-bom** [comment]
Thanks @MoseleyS!

**Pull request merged**